compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -182,15 +182,17 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     self.arena.alloc_from_iter(arms.iter().map(|x| self.lower_arm(x))),
                     hir::MatchSource::Normal,
                 ),
-                ExprKind::Async(capture_clause, closure_node_id, block) => self.make_async_expr(
-                    *capture_clause,
-                    hir_id,
-                    *closure_node_id,
-                    None,
-                    e.span,
-                    hir::AsyncGeneratorKind::Block,
-                    |this| this.with_new_scopes(|this| this.lower_block_expr(block)),
-                ),
+                ExprKind::Async(capture_clause, closure_node_id, block) => {
+                    return self.make_async_expr(
```

Comment:
I think the ICE is because of this -- why don't you make `make_async_expr` return an `ExprKind` and go thru the regular codepaths? The unused HIR id is on line 69